mu-plugins/10up-plugin/includes/classes/PostTypes/AbstractPostType.php

```diff
@@ -0,0 +1,187 @@
+		$labels = array(
+			'name'                     => $plural_label,
+			// Already translated via get_plural_label().
+			'singular_name'            => $singular_label,
+			// Already translated via get_singular_label().
+			'add_new_item'             => sprintf( __( 'Add New %s', 'tenup-plugin' ), $singular_label ),
+			'edit_item'                => sprintf( __( 'Edit %s', 'tenup-plugin' ), $singular_label ),
+			'new_item'                 => sprintf( __( 'New %s', 'tenup-plugin' ), $singular_label ),
+			'view_item'                => sprintf( __( 'View %s', 'tenup-plugin' ), $singular_label ),
+			'view_items'               => sprintf( __( 'View %s', 'tenup-plugin' ), $plural_label ),
+			'search_items'             => sprintf( __( 'Search %s', 'tenup-plugin' ), $plural_label ),
+			'not_found'                => sprintf( __( 'No %s found.', 'tenup-plugin' ), strtolower( $plural_label ) ),
+			'not_found_in_trash'       => sprintf( __( 'No %s found in Trash.', 'tenup-plugin' ), strtolower( $plural_label ) ),
+			'parent_item_colon'        => sprintf( __( 'Parent %s:', 'tenup-plugin' ), $plural_label ),
+			'all_items'                => sprintf( __( 'All %s', 'tenup-plugin' ), $plural_label ),
+			'archives'                 => sprintf( __( '%s Archives', 'tenup-plugin' ), $singular_label ),
+			'attributes'               => sprintf( __( '%s Attributes', 'tenup-plugin' ), $singular_label ),
+			'insert_into_item'         => sprintf( __( 'Insert into %s', 'tenup-plugin' ), strtolower( $singular_label ) ),
+			'uploaded_to_this_item'    => sprintf( __( 'Uploaded to this %s', 'tenup-plugin' ), strtolower( $singular_label ) ),
+			'filter_items_list'        => sprintf( __( 'Filter %s list', 'tenup-plugin' ), strtolower( $plural_label ) ),
+			'items_list_navigation'    => sprintf( __( '%s list navigation', 'tenup-plugin' ), $plural_label ),
+			'items_list'               => sprintf( __( '%s list', 'tenup-plugin' ), $plural_label ),
+			'item_published'           => sprintf( __( '%s published.', 'tenup-plugin' ), $singular_label ),
+			'item_published_privately' => sprintf( __( '%s published privately.', 'tenup-plugin' ), $singular_label ),
+			'item_reverted_to_draft'   => sprintf( __( '%s reverted to draft.', 'tenup-plugin' ), $singular_label ),
+			'item_scheduled'           => sprintf( __( '%s scheduled.', 'tenup-plugin' ), $singular_label ),
+			'item_updated'             => sprintf( __( '%s updated.', 'tenup-plugin' ), $singular_label ),
+			'menu_name'                => $plural_label,
+			'name_admin_bar'           => $singular_label,
+		);
+		// phpcs:enable
+
+		return $labels;
+	}
+
+	/**
+	 * Registers a post type and associates its taxonomies.
+	 *
+	 * @uses $this->get_name() to get the post's type name.
+	 * @return Bool Whether this theme has supports for this post type.
+	 */
+	public function register() {
+		$this->register_post_type();
+		$this->register_taxonomies();
```

Comment:
@fabiankaegy so the abstract post type already has an `after_register()` method that can be used to do anything you need after the post type has been registered, including registering meta fields. Do you think that, with some documentation would suffice, or are you thinking a dedicated meta registration method?

If the latter, how are you picturing it working? 

The other option is that we could utilise a PHP trait for meta registration that would allow us to share the meta registration method across post types and taxonomies, whilst sharing a common API 🤔

Reply:
I'm not sure what the best answer is here. I recognize that meta isn't necessarily easy because as you mention it may need to be registered for multiple post types etc. 

I'd almost like to come up with a way to handle meta registration in a separate way kind of like taxonomies or even blocks. Because in the end a piece of meta really is both registering the field but then also creating the UI for it likely in React these days. So it's kind or it's own module (part of a feature)... 

I'm not even sure we need to standardize it now. But I see many projects doing it in many different ways which I think eliminating this is one of the goals we have here. 

We can merge this with some docs for how you do it now and then maybe revisit in the future :) 

My goal is it for us to make it so easy that we finally can convince folks to stop loading field manager =D

Reply:
@fabiankaegy I _really_ like the idea of meta being shipped with the JS needed to register it. I wonder if there's some way that we can build some default react interfaces that can be extended 🤔 

Can you open an issue that we can hash these ideas out in and we'll leave this PR as-is for now. I don't want to over-complicate it too much or make it _too_ abstract before we've got anything merged.

I'll add docs around the usage of this code and how to add post meta etc 🙂